skel/daemons/settings.go

```diff
@@ -0,0 +1,368 @@
+								updateWallet(peatioClient, w, funk.UniqString(mergedCurrencies))
+							}
+						}
+					}
+				} else if walletKey == "none" {
+					createDepositAndHotWallet(peatioClient, currencyGroup, opendaxAddr)
+				}
+			}
+		}
+	}
+}
+
+// Method divideCurrenciesIntoGroups divides currencies into groups
+// In first place system always puts parent ID
+// For example {"eth": ["eth", "usdt", "link"]}
+func divideCurrenciesIntoGroups(currencies []CurrencyResponse) map[string][]string {
+	res := make(map[string][]string)
+	// Sort currencies with empty Parent ID first
+	// It means those currencies should have separate wallet
+	sort.SliceStable(currencies, func(i, j int) bool {
+		return currencies[i].ParentID < currencies[j].ParentID
+	})
+
+	for _, currency := range currencies {
+		if currency.Type == "coin" {
+			if currency.ParentID == "" {
+				res[currency.ID] = append(res[currency.ID], currency.ID)
+			} else {
+				res[currency.ParentID] = append(res[currency.ParentID], currency.ID)
+			}
+		}
+	}
+
+	return res
+}
+
+// Method createDepositAndHotWallet creates deposit and hot wallet
+func createDepositAndHotWallet(peatioClient *peatio.Client, currencies []string, opendaxAddr string) {
+	// Create Deposit Wallet
+	depositWalletParams := peatio.CreateWalletParams{
+		BlockchainKey: "opendax-cloud",
+		Name:          fmt.Sprintf("%s Deposit Wallet", strings.ToUpper(currencies[0])),
+		Kind:          "deposit",
+		Gateway:       "opendax_cloud",
+		Address:       "address",
+		Currencies:    currencies,
+		Status:        "active",
+		Settings: peatio.Settings{
+			URI: fmt.Sprintf("%v/api/v2/opx/peatio", opendaxAddr),
+		},
+	}
+
+	_, depositApiError := peatioClient.CreateWallet(depositWalletParams)
+	if depositApiError != nil {
+		log.Printf("ERROR: createWallets: Can't create deposit wallet. Error: %v. Errors: %v", depositApiError.Error, depositApiError.Errors)
+		return
+	}
+
+	// Create Hot Wallet
+	hotWalletParams := peatio.CreateWalletParams{
+		BlockchainKey: "opendax-cloud",
+		Name:          fmt.Sprintf("%s Hot Wallet", strings.ToUpper(currencies[0])),
+		Kind:          "hot",
+		Gateway:       "opendax_cloud",
+		Address:       "address",
+		Currencies:    currencies,
+		Status:        "active",
+		Settings: peatio.Settings{
+			URI: fmt.Sprintf("%v/api/v2/opx/peatio", opendaxAddr),
+		},
+	}
+
+	_, hotApiError := peatioClient.CreateWallet(hotWalletParams)
+	if hotApiError != nil {
+		log.Printf("ERROR: createWallets: Can't create deposit wallet. Error: %v. Errors: %v", hotApiError.Error, hotApiError.Errors)
+		return
+	}
+}
+
+// Method findCurrenciesInWallets finds wallet by set of currencies
+func findCurrenciesInWallets(wallets []*peatio.Wallet, currencies []string) map[string][]*peatio.Wallet {
+	res := map[string][]*peatio.Wallet{}
+
+	// Sort currencies to make sure it will be in right order with wallet currencies
+	newCurrencies := make([]string, len(currencies))
+	copy(newCurrencies, currencies)
```

Comment:
Since `currencies` is not a pointer, why do we need a `copy` instead of just reusing `currencies`?

Reply:
Well, somehow in Sort method it changes currencies group directly (but we need exactly same order as it was lately, when we create wallets)